engine/api/v2_workflow_run_engine.go

```diff
@@ -0,0 +1,384 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/go-gorp/gorp"
+	"github.com/rockbears/log"
+
+	"github.com/ovh/cds/engine/api/rbac"
+	"github.com/ovh/cds/engine/api/region"
+	"github.com/ovh/cds/engine/api/user"
+	"github.com/ovh/cds/engine/api/workflow_v2"
+	"github.com/ovh/cds/engine/cache"
+	"github.com/ovh/cds/sdk"
+	"github.com/ovh/cds/sdk/telemetry"
+)
+
+func (api *API) V2WorkflowRunEngineChan(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
+			if ctx.Err() != nil {
+				return ctx.Err()
+			}
+			return nil
+		case wrEnqueue := <-api.workflowRunTriggerChan:
+			if err := api.workflowRunV2Trigger(ctx, wrEnqueue); err != nil {
+				log.ErrorWithStackTrace(ctx, err)
+			}
+		}
+	}
+}
+
+func (api *API) V2WorkflowRunEngineDequeue(ctx context.Context) error {
+	for {
+		var wrEnqueue sdk.V2WorkflowRunEnqueue
+		if err := api.Cache.DequeueWithContext(ctx, workflow_v2.WorkflowEngineKey, 250*time.Millisecond, &wrEnqueue); err != nil {
+			log.Error(ctx, "V2WorkflowRunEngine > DequeueWithContext err: %v", err)
+			continue
+		}
+		if err := api.workflowRunV2Trigger(ctx, wrEnqueue); err != nil {
+			log.ErrorWithStackTrace(ctx, err)
+		}
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+	}
+}
+
+// TODO Manage stage
+// TODO Manage job sub number
+// TODO manage git context
+// TODO manage vars context
+func (api *API) workflowRunV2Trigger(ctx context.Context, wrEnqueue sdk.V2WorkflowRunEnqueue) error {
+	_, next := telemetry.Span(ctx, "api.workflowRunV2Trigger.lock")
```

Comment:
It could be nice to be able to continue the global span from the handler

Reply:
It will be made in the next PR